automation-tests/test/utilities.ts

```diff
@@ -0,0 +1,337 @@
+    await browser.keys([CMD_KEY, 'a', 'c']);
+    // Should be able to use Keys.Ctrl, but Keys is not exported from webdriverio
+    // See https://webdriver.io/docs/api/browser/keys/
+    const terminalText = await clipboard.read();
+
+    if (terminalText && terminalText !== '') {
+      return terminalText
+    }
+  }
+
+  throw new Error('Exceeded time limit - text in the terminal was not found');
+}
+
+async function getFilteredVisibleTreeViewItems(workbench: Workbench, projectName: string, searchString: string): Promise<ViewItem[]> {
+  const sidebar = workbench.getSideBar();
+  const treeViewSection = await sidebar.getContent().getSection(projectName);
+  await treeViewSection.expand();
+
+  // Warning, we can only retrieve the items which are visible.
+  const visibleItems = (await treeViewSection.getVisibleItems()) as DefaultTreeItem[];
+  const filteredItems = (await visibleItems.reduce(async (previousPromise: Promise<ViewItem[]>, currentItem: ViewItem) => {
+    const results = await previousPromise;
+    const label = await (currentItem as TreeItem).getLabel();
+    if (label.startsWith(searchString)) {
+      results.push(currentItem);
+    }
+
+    return results;
+  }, Promise.resolve([])) as ViewItem[]);
+
+  return filteredItems;
+}
+
+async function getFilteredVisibleTreeViewItemLabels(workbench: Workbench, projectName: string, searchString: string): Promise<string[]> {
+  const sidebar = workbench.getSideBar();
+  const treeViewSection = await sidebar.getContent().getSection(projectName);
+  await treeViewSection.expand();
+
+  // Warning, we can only retrieve the items which are visible.
+  const visibleItems = (await treeViewSection.getVisibleItems()) as DefaultTreeItem[];
+  const filteredItems = (await visibleItems.reduce(async (previousPromise: Promise<string[]>, currentItem: ViewItem) => {
+    const results = await previousPromise;
+    const label = await (currentItem as TreeItem).getLabel();
+    if (label.startsWith(searchString)) {
+      results.push(label);
+    }
+
+    return results;
+  }, Promise.resolve([])) as string[]);
+
+  return filteredItems;
+}
+
+function currentUserName(): string {
+  const userName = os.userInfo().username ||
+    process.env.SUDO_USER ||
+    process.env.C9_USER ||
+    process.env.LOGNAME ||
+    process.env.USER ||
+    process.env.LNAME ||
+    process.env.USERNAME;
+
+  return userName;
+}
+
+export const utilities = {
+  openFile,
+  openFolder,
+  createFolder,
+  removeFolder,
+  pause,
+  log,
+  clickFilePathOkButton,
+  openCommandPromptWithCommand,
+  executeQuickPick,
+  selectQuickPickItem,
+  getStatusBarItemWhichIncludes,
+  waitForNotificationToGoAway,
+  notificationIsPresent,
+  attemptToFindNotification,
+  selectChannel,
+  getOutputPanelText,
+  attemptToFindOutputPanelText,
+  executeCommand,
+  getTerminalView,
+  getTerminalViewText,
+  getFilteredVisibleTreeViewItems,
+  getFilteredVisibleTreeViewItemLabels,
+  currentUserName,
```

Comment:
Hum..this list already seems like quite the mixed bag.  Have you considered splitting this out into separate modules that have a specific purpose? It'd make them easier to find. Maybe a `utilities` directory with modules like, terminal, vscode, logger, folder (or filesystem)